### ISSUE TITLE
module_function does not disable Utility Function

### ISSUE THREAD

**Issue by ashmoran** [troessner/reek#27]
A bit like a method on a class, the following should presumably not trigger Utility Function?

    module MyModule
      def reset!
        # ...
      end
      module_function :reset!
    end

**Comment by troessner** [comment]
I'll look into this after the reek 2 release.

**Comment by troessner** [comment]
I had hoped that fixing #112 would fix this one as well, but it isn't:

```
  ( $ (develop)) cat /tmp/foo.rb 
```

with

```Ruby
class C
  def m; 3 + 7; end
  module_function :m
end
```

gives:

```Bash
  ( $ (develop)) bin/reek /tmp/foo.rb --smell UtilityFunction
/tmp/foo.rb -- 1 warning:
  [2]:C#m doesn't depend on instance state (UtilityFunction)
```

The AST looks like this:

```
[1] pry(main)> src = 'class C; def m; 3 + 7; end; module_function :m; end'
=> "class C; def m; 3 + 7; end; module_function :m; end"
[2] pry(main)> require 'parser/current'
=> true
[3] pry(main)> parser = Parser::CurrentRuby
=> Parser::Ruby22
[4] pry(main)> parser.parse src
=> (class
  (const nil :C) nil
  (begin
    (def :m
      (args)
      (send
        (int 3) :+
        (int 7)))
    (send nil :module_function
      (sym :m))))
```

So we'll need to fix this separately.

**Comment by troessner** [comment]
Started to work on this in #541

**Comment by troessner** [comment]
And now working on it in #567

**Comment by troessner** [comment]
Finally, 6 years after this bug report this has been fixed in #567 .:)
